go-controller/pkg/cni/types.go

```diff
@@ -142,6 +148,14 @@ type PodRequest struct {
 	ctx context.Context
 	// cancel should be called to cancel this request
 	cancel context.CancelFunc
+
+	// network information associated with the given netconf
+	netInfo util.NetInfo
```

Comment:
maybe I am missing something, how do ovn-cni knows it is the first attachement or not? does that information embedded somewhere?

Reply:
Well the most reasonable way would be to check the network name. `default` or empty should only happen for the primary attachment and any other value for secondary attachments.

Alternatively, we can check if the primary interface already exists in the host and/or in the pod (probably the pod is a better place to check) on CmdAdd. If it doesn't then it means that CmdAdd is the primary attachment.

If we cannot rely on any of those two, I think it would be much better to have an additional paramater in Netconf that would be the network name instead of relying on (an unreliable) Netconf name as the name for the network.